### PR TITLE
Uses OpenAPI v3 for Cosmos client

### PR DESCRIPTION
**Pull request by ArmandGrillet**
This introduces a dcos.yaml spec file based on the OpenAPI version 3.

For now only the Cosmos autogenerated client is in the example.go.

The Secrets spec in dcos.yaml will need to be slightly modified to lead
to a functional client.

IAM and marathon still need to integrated.

Co-authored-by: Jimmi Dyson <jimmidyson@gmail.com>
Co-authored-by: Thorsten Ball <mrnugget@gmail.com>
Co-authored-by: Armand Grillet <armand.grillet@outlook.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,4 +12,8 @@ lint: fmt
 
 .PHONY: fmt
 fmt:
-	bash -c "diff -u <(echo -n) <(gofmt -d -s .)"
+	gofmt -w -s .
+
+.PHONY: generate
+generate:
+	openapi-generator generate -i openapi/dcos.yaml -g go -o dcos --skip-validate-spec -D packageName=dcos,withGoCodegenComment=true -t templates
```
Comment (jimmidyson) [inline Makefile]: Can this use the docker image instead?

Comment (jimmidyson) [inline Makefile]: Not sure we want to `--skip-validate-spec` feels like that's quite important for us while we stabilize the spec.

Comment (jimmidyson) [inline Makefile]: We shouldn't generate all the files, they're not necessary IMO. I had to specify only the files I wanted to generate in separate runs of `generate` command - see https://github.com/dcos/client-go/pull/13/files#diff-06f2134279436229c113632db3603d73R30 for an example. This means we don't have the travis file generated, etc. which is just cruft for the repo.

Comment (bamarni) [reply Makefile]: I agree, I added it to be able to get something generated, as there are some missing description in the specs

Comment (mrnugget) [reply Makefile]: Done

**Comment by jimmidyson** [comment]
This PR dropped package search from #10... Adding back in in another PR...

**Comment by hectorj2f** [comment]
I believe it might be hard to review 402 file changes :S

**Pull request merged**